gcore/gdalalgorithm.cpp

```diff
@@ -2215,6 +2215,20 @@ bool GDALAlgorithm::ProcessDatasetArg(GDALAlgorithmArg *arg,
         if (osDatasetName == "-" && (flags & GDAL_OF_UPDATE) == 0)
             osDatasetName = "/vsistdin/";
 
+        if ((flags & GDAL_OF_UPDATE) != 0 && m_callPath.size() == 4 &&
```

Comment:
```suggestion
        // Handle special case of overview delete in GTiff which would fail 
        // if it is COG without IGNORE_COG_LAYOUT_BREAK
        if ((flags & GDAL_OF_UPDATE) != 0 && m_callPath.size() == 4 &&
```